iroh/src/commands.rs

```diff
@@ -194,10 +190,16 @@ pub enum Commands {
     /// Fetch the data identified by HASH from a provider
     Get {
         /// The hash to retrieve, as a Blake3 CID
-        hash: Blake3Cid,
+        #[clap(conflicts_with = "ticket", required_unless_present = "ticket")]
+        hash: Option<Blake3Cid>,
         /// PeerId of the provider
-        #[clap(long, short)]
-        peer: PeerId,
+        #[clap(
+            long,
+            short,
+            conflicts_with = "ticket",
+            required_unless_present = "ticket"
```

Comment:
whoa, this allows you to do complex things.  nice

Reply:
I had chatgpt explain it to me. There are still some downsides, e.g. the usage message is a bit weird.